### ISSUE TITLE
A warning past an interpolation ends two columns short under `postcss-styled-syntax`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#644]
### What is the problem you're trying to solve?

Under `postcss-styled-syntax` a warning whose `endIndex` lies past a `${…}` interpolation ends short of where it points, by two columns per interpolation. Measured on `f4048da` with `declaration-block-single-line-max-declarations: 1`:

```js
const A = styled.div`a { color: ${c}; top: 0; }`
// reported 1:24 to 1:48; the closing brace stands at column 48, so the span ends in front of it
const A = styled.div`a { color: ${c}; top: ${d}; }`
// reported 1:24 to 1:51; the brace stands at column 56
const A = styled.div`a { color: pink; top: 0; }`
// reported 1:24 to 1:48, the brace at column 48: right without an interpolation
```

The same shortfall stands in front of the brace for `block-closing-brace-newline-before: "always"` and `block-closing-brace-space-before: "never"`, which report `1:46` over the first template. The text the rules read is the file's: `blockString` and `nodeString` print `{ color: ${c}; top: 0; }` character for character, so the index the rule reports is right and what maps it to a column counts the interpolation shorter than the file spells it. `block-opening-brace-newline-after` reports `1:25` there, in front of the interpolation, and is right.

### What solution would you like to see?

A position past an interpolation lands where the file spells it, whether the syntax's mapping of an index to a column or the plugin's reading of it has to change. Found while writing the fixer of #641; the case in `lib/syntaxes/styled/rules/declaration-block-single-line-max-declarations/index.test.ts` pins the column as the base reports it and names this issue.
